fabric-data-generation-api-v1/build.gradle

```diff
@@ -0,0 +1,109 @@
+archivesBaseName = "fabric-data-generation-api-v1"
+version = getSubprojectVersion(project)
+
+moduleDependencies(project, [
+		'fabric-api-base',
+		'fabric-registry-sync-v0',
+		'fabric-networking-api-v1'
+])
+
+sourceSets {
+	testmod {
+		java {
+			srcDirs += [
+				'src/testmod/generated'
+			]
+		}
+	}
+}
+
+loom {
+	accessWidenerPath = file("src/main/resources/fabric-data-generation-api-v1.accesswidener")
+
+	runs {
+		datagen {
+			inherit testmodServer
+			name "Data Generation"
+			vmArg "-Dfabric-api.datagen"
+			vmArg "-Dfabric-api.datagen.output-dir=${file("src/testmod/generated")}"
+			vmArg "-Dfabric-api.datagen.strict-validation"
+
+			ideConfigGenerated = true
+			runDir "build/datagen"
+		}
+		datagenClient {
+			client()
+			name "Data Generation"
+			vmArg "-Dfabric-api.datagen"
+			vmArg "-Dfabric-api.datagen.output-dir=${file("src/testmod/generated")}"
+			vmArg "-Dfabric-api.datagen.strict-validation"
+
+			ideConfigGenerated = true
+			runDir "build/datagen"
+			source sourceSets.testmod
+		}
+	}
+}
+
+assemble.dependsOn runDatagen
+
+import org.objectweb.asm.ClassReader
+import org.objectweb.asm.Opcodes
+import org.objectweb.asm.tree.ClassNode
+
+import java.util.zip.ZipFile
+
+task generateAccessWidener() {
+	doLast {
+		File inputJar = loom.mappingsProvider.mappedProvider.baseMappedJar
```

Comment:
Also, is Fabric API setting a bad example to modders here by accessing loom internals? 😛

Reply:
Yes, but this is not an example ;) I might put a comment on it.